lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1669,9 +1668,49 @@ CompilerType
 TypeSystemSwiftTypeRef::GetPointeeType(opaque_compiler_type_t type) {
   return m_swift_ast_context->GetPointeeType(ReconstructType(type));
 }
+
 CompilerType
 TypeSystemSwiftTypeRef::GetPointerType(opaque_compiler_type_t type) {
-  return m_swift_ast_context->GetPointerType(ReconstructType(type));
+  auto impl = [&]() -> CompilerType {
+    using namespace swift::Demangle;
+    Demangler dem;
+
+    auto *pointee_type = GetDemangledType(dem, AsMangledName(type));
+
+    auto *pointer_type = dem.createNode(Node::Kind::Type);
+    auto *BGS = dem.createNode(Node::Kind::BoundGenericStructure);
+    pointer_type->addChild(BGS, dem);
+
+    NodePointer parent;
+    NodePointer child;
+
+    // Construct the first (Type) branch of BoundGenericStructure.
+    parent = BGS;
```

Comment:
Personally I find it difficult to keep track of what parent and child are. What do you think about using new local vairable names for each node (cf. line 206ff) and perhaps even nested scopes to reflect the nesting structure of the nodes?

Reply:
will do, it's nicer.